### ISSUE TITLE
Patch request giving internal server error.

### ISSUE THREAD

**Issue by Subhojit-Dey1234** [centerofci/mathesar#2179]
## Description
Changing the title of the Schemas Title gives internal server error ( Status code - 500 )

## Expected behavior
Should return 200 status code. The title is updated after refreshing the page.

## To Reproduce
Try to edit the title of the schemas.
![image](https://user-images.githubusercontent.com/66687621/210250962-6a9c84de-6465-4b2a-a6c9-396cca432595.png)
![ezgif-2-670dc6a67d](https://user-images.githubusercontent.com/66687621/210252610-7f66839a-5cbd-4c20-9e2b-23bab6315adb.gif)

## Environment
 - OS: Windows
 - Browser: Brave, Chrome

**Comment by kgodey** [comment]
@Subhojit-Dey1234 could you please provide the Python traceback of the error? If you open the error in a new window, you should see an HTML page. Above the traceback, there should be a "switch to copy paste view" link which will give you the traceback to add here.

**Comment by Subhojit-Dey1234** [comment]
@kgodey I am attaching the screenshots for the traceback.
![image](https://user-images.githubusercontent.com/66687621/210396926-32ed98d3-431f-49e3-9531-c10627db67e5.png)
![image](https://user-images.githubusercontent.com/66687621/210397016-b0cf8a10-bf69-4fe3-9307-f757bb0bbc5e.png)
![image](https://user-images.githubusercontent.com/66687621/210397080-f4b69995-dc78-4bb3-9754-0f75718c8520.png)
![image](https://user-images.githubusercontent.com/66687621/210397287-c0d9e276-84b5-4960-aaf5-d2b9328db5dd.png)

**Comment by kgodey** [comment]
@Subhojit-Dey1234 thanks for the screenshots. It would be easier if we had the pasted traceback, here's the button that you can use to get the copy/pastable view. Let me know if that doesn't work for some reason.

![error_movies_not_imported_in_views](https://user-images.githubusercontent.com/287034/210425703-362fb7da-18d6-4cd7-9d98-ddb12bf1b87e.png)

**Comment by Subhojit-Dey1234** [comment]
The button was previously not working. I guess this is the thing.
```
Environment:


Request Method: PATCH
Request URL: http://localhost:8000/api/db/v0/schemas/4/

Django Version: 3.1.14
Python Version: 3.9.16
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
 'django_filters',
 'django_property_filter',
 'mathesar']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'mathesar.middleware.CursorClosedHandlerMiddleware',
 'mathesar.middleware.PasswordChangeNeededMiddleware',
 'django_userforeignkey.middleware.UserForeignKeyMiddleware',
 'django_request_cache.middleware.RequestCacheMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 466, in handle_exception
    response = exception_handler(exc, context)
  File "/code/mathesar/exception_handlers.py", line 59, in mathesar_exception_handler
    raise exc
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/code/mathesar/api/db/viewsets/schemas.py", line 47, in partial_update
    schema.update_sa_schema(serializer.validated_data)
  File "/code/mathesar/models/base.py", line 207, in update_sa_schema
    reset_reflection()
  File "/code/mathesar/state/base.py", line 27, in reset_reflection
    _trigger_django_model_reflection()
  File "/code/mathesar/state/base.py", line 31, in _trigger_django_model_reflection
    reflect_db_objects(metadata=get_cached_metadata())
  File "/code/mathesar/state/django.py", line 47, in reflect_db_objects
    reflect_columns_from_tables(tables, metadata=metadata)
  File "/code/mathesar/state/django.py", line 119, in reflect_columns_from_tables
    _delete_stale_columns(attnum_tuples, tables)
  File "/code/mathesar/state/django.py", line 160, in _delete_stale_columns
    stale_columns_query = reduce(

Exception Type: TypeError at /api/db/v0/schemas/4/
Exception Value: reduce() of empty sequence with no initial value

```

**Comment by kgodey** [comment]
Thanks @Subhojit-Dey1234

**Comment by Subhojit-Dey1234** [comment]
Can you help me with what will be in the array??? I can give it a try.

**Comment by kgodey** [comment]
@Subhojit-Dey1234 which array are you referring to?

**Comment by Subhojit-Dey1234** [comment]
The error is mainly because `stale_columns_conditions` is empty and in reduce function we can't pass an empty array.

**Comment by Subhojit-Dey1234** [comment]
Here the `attnum_tuples` are coming out empty and it is causing the main problem. Can you tell what should be in this tuple?
![image](https://user-images.githubusercontent.com/66687621/211078517-46e5595c-5702-41a6-8067-413c361fa279.png)

**Comment by silentninja** [comment]
@Subhojit-Dey1234 This error will occur when a Table does not have any columns. A table created by mathesar always contains an `id` column. Can you confirm if you have a Table with no columns that wasn't created by mathesar? This would help us understand if some other issue led to creation of a Table with no columns.

In order to fix this issue, we should supply an empty array using the `initial` parameter as an initializer to the `reduce` function. Let me know if you are interested to send a PR for this issue.

**Comment by Subhojit-Dey1234** [comment]
Yes I tried by creating a schema then editing the title of the schema.
Here the error was popping.

**Comment by silentninja** [comment]
Closed by #2229
